spec/models/lettings_log_spec.rb

```diff
@@ -1454,6 +1454,15 @@ def previous_postcode_fields
         expect(record_from_db["vacdays"]).to eq(0)
         expect(lettings_log["vacdays"]).to eq(0)
       end
+
+      it "correctly derives and saves first_time_property_let_as_social_housing" do
+        record_from_db = ActiveRecord::Base.connection.execute(
+          "select first_time_property_let_as_social_housing" \
+          " from lettings_logs where id=#{lettings_log.id}",
+        ).to_a[0]
```

Comment:
why do we hand craft SQL queries when we have active record?

```suggestion
        record_from_db = LettingsLog.find(lettings_log.id)
```

Reply:
you can add a `select` if you really want to restrict the data we request and that comes back

Reply:
well at this moment in time i'm gonna say the rest of the file is wrong. we don't interface with the database directly with SQL queries it goes through active record. therefore validating the database has the correct data with SQL, only to then interface with it through active record seems absurd

Reply:
Agreed!

Reply:
Context to why it is the way it is: 
At some point we weren't saving the inferred values to the db, but we had methods that would return that value on the model. After that we made the decision that we actually wanted to persist the value to the db. This code was a consequence of basically testing that.
If we had it as `expect(LettingsLog.find(lettings_log.id).field_we_are_testing_for).to eq(expected_value)` it wold call the method on the model and wouldn't fail as expected
I don't love how this test looks, but in terms of confidence I think it is testing the fact that the data is actually persisted and I'm not sure how else we could do that

Reply:
We have a ticket now so at least someone can take a look at it. i think at least a few of us thing its less than ideal